Add optional dependencies (#4480)
pydata/xarray: doc/installing.rst

    $ pip install xarray

We also maintain other dependency sets for different subsets of functionality::

>     $ pip install "xarray[io]"        # Install requirements for handling I/O
>     $ pip install "xarray[accel]"     # Install requirements for accelerating xarray
>     $ pip install "xarray[parallel]"  # Install requirements for dask arrays
>     $ pip install "xarray[viz]"       # Install requirements for visualization

Reviewer: ```suggestion
    $ pip install "xarray[io]"        # Install optional dependencies for handling I/O
    $ pip install "xarray[accel]"     # Install optional dependencies for accelerating xarray
    $ pip install "xarray[parallel]"  # Install optional dependencies for dask arrays
    $ pip install "xarray[viz]"       # Install optional dependencies for visualization
```

not sure about this one

    $ pip install "xarray[complete]"  # Install all the above

The above commands should install most of the `optional dependencies`_. However,

> some packages which are either not listed in the PyPI or require extra
> libraries are omitted and would require extra installation steps.

Reviewer: ```suggestion
some packages which are either not listed on PyPI or would require extra
installation steps are omitted.
```

> 

Author: @mathause Done.

    $ pip install xarray

We also maintain other dependency sets for different subsets of functionality::

    $ pip install "xarray[io]"        # Install optional dependencies for handling I/O
    $ pip install "xarray[accel]"     # Install optional dependencies for accelerating xarray
    $ pip install "xarray[parallel]"  # Install optional dependencies for dask arrays
    $ pip install "xarray[viz]"       # Install optional dependencies for visualization
    $ pip install "xarray[complete]"  # Install all the above

The above commands should install most of the `optional dependencies`_. However,
some packages which are either not listed on PyPI or require extra
installation steps are omitted. To know which dependencies would be installed,

> take a look at the ``[options.extras_require]`` section in setup.cfg_.

Reviewer 2: I don't think we should link to the `stable` docs because this may be out-of-date for other branches. Maybe we can use the `literalinclude` directive?

Author: Great suggestion. Done.
